pallets/reservable-currency/src/lib.rs

```diff
@@ -91,11 +94,15 @@ decl_module! {
 		) -> DispatchResult {
 			let _ = ensure_signed(origin)?; // dangerous because can be called with any signature (so dont do this in practice ever!)
 
+                        // Unreserved contains our overdraft, if collateral is bigger than
+                        // to_punish's reserved balance, what's left is put in unreserved.
 			let unreserved = T::Currency::unreserve(&to_punish, collateral);
-			T::Currency::transfer(&to_punish, &dest, unreserved, AllowDeath)?;
+                        // Perhaps add an aditional check to see if we can susbtract unreserved
+                        // from free_balance and add it into the transfer below.
```

Comment:
No additional check is necessary in this case. We know `to_punish` will have enough free balance because we _just_ unreserved it.

That being said, you're thinking about this exactly the right way. Always good to be watching out for over- and under-flows.

Reply:
I see, I think I worded the remark wrongly. I guess it's dependant on implementation needs, but say that the target's ``` reserved_balance ``` is smaller than the ``` collateral ```, what we're doing now is ``` collateral - overdraft```. What I was thinking off is seeing if instead of just subtracting it like we're doing above, perhaps just try and see if there's additional ``` unreserved_balance ``` available to take the "overdraft" from. Not sure if that makes sense, but as I said, most likely implementation specific? Thanks you!

Reply:
Oh, I see what you mean. Yeah the design you described is totally reasonable too.